llvm/lib/Target/X86/Disassembler/X86Disassembler.cpp

```diff
@@ -636,14 +664,13 @@ static int readModRM(struct InternalInstruction *insn) {
     break;
   }
 
-  reg |= rFromREX(insn->rexPrefix) << 3;
-  rm |= bFromREX(insn->rexPrefix) << 3;
+  reg |= (rFromREX(insn->rexPrefix) << 3) |
+         (r2FromREX2(insn->rex2ExtensionPrefix[1]) << 4);
+  rm |= (bFromREX(insn->rexPrefix) << 3) |
+        (b2FromREX2(insn->rex2ExtensionPrefix[1]) << 4);
```

Comment:
Do we also need to `| xFromEVEX2of4(insn->vectorExtensionPrefix[1]) << 4`

Reply:
No need. As I said in the description, we "Simulate the REX2 prefix w/ the 2nd and 3rd payloads of EVEX"